### ISSUE TITLE
Hanging compile with operator overloading

### ISSUE THREAD

**Issue by icager** [manifold-systems/manifold#132]
**Describe the bug**
When manifold-ext is included in a project, any expression with more than ~20 variables `+`'d together results in the compiler hanging.

**To Reproduce**
Using the following gradlescript:
```
apply plugin: 'java'

repositories {
    jcenter()
    mavenCentral()
    maven { url 'https://oss.sonatype.org/content/repositories/snapshots/' }
}

dependencies {
    compile group: 'systems.manifold', name: 'manifold-graphql', version: '2019.1.22'
    annotationProcessor group: 'systems.manifold', name: 'manifold-graphql', version: '2019.1.22'
}

tasks.withType(JavaCompile) {
    options.compilerArgs += ['-Xplugin:Manifold', '-verbose']
}

tasks.compileJava {
    classpath += files(sourceSets.main.output.resourcesDir)
    dependsOn processResources
}
tasks.compileTestJava {
    classpath += files(sourceSets.test.output.resourcesDir)
    dependsOn processTestResources
}
```
and the following Java class:
```java
public class LocalVarTest {

    public String returnAString() {
        final String a1 = "a";
        final String a2 = "a";
        final String a3 = "a";
        final String a4 = "a";
        final String a5 = "a";
        final String a6 = "a";
        final String a7 = "a";
        final String a8 = "a";
        final String a9 = "a";
        final String a10 = "a";
        final String a11 = "a";
        final String a12 = "a";

        final String b1 = "a";
        final String b2 = "a";
        final String b3 = "a";
        final String b4 = "a";
        final String b5 = "a";
        final String b6 = "a";
        final String b7 = "a";
        final String b8 = "a";
        final String b9 = "a";
        final String b10 = "a";
        final String b11 = "a";
        final String b12 = "a";

        return a1 + a2 + a3 + a4 + a5 + a6 + a7 + a8 + a9 + a10 + a11 + a12 +
            b1 + b2 + b3 + b4 + b5 + b6 + b7 + b8 + b9 + b10 + b11 + b12;
    }

}
```
then building with `./gradlew clean build` hangs.

The build completes fine if the values are written inline:
```java
return "a" + "a" + "a" + "a" + ...
```

**Expected behavior**
Compilation should complete.

**Desktop (please complete the following information):**
```
------------------------------------------------------------
Gradle 4.10.2
------------------------------------------------------------

Build time:   2018-09-19 18:10:15 UTC
Revision:     b4d8d5d170bb4ba516e88d7fe5647e2323d791dd

Kotlin DSL:   1.0-rc-6
Kotlin:       1.2.61
Groovy:       2.4.15
Ant:          Apache Ant(TM) version 1.9.11 compiled on March 23 2018
JVM:          1.8.0_232 (Oracle Corporation 25.232-b09)
OS:           Linux 4.9.93-linuxkit-aufs amd64
```

**Additional context**
I came across this while trying to use manifold-graphql in a project that also uses Lombok. There are several large data classes with a lot of fields, for which Lombok's `@ToString` annotation generates methods like the above.

Adding '-verbose' to the compiler flags shows the compiler hanging on these classes. Triggering a few thread dumps in the compiler suggests that things are getting stuck in `ManAttr.handleOperatorOverloading`:
```
Type left = (Type)checkNonVoid.invoke( tree.lhs.pos(), attribExpr.invoke( tree.lhs, getEnv() ) );
```

I'm guessing it's just running in exponential time rather than completely hanging.

If fixing this problem directly isn't easy (I'm not familiar enough with compiler plugins to guess what might be involved) then the ability to disable operator overloading (or have graphql not pull it in) either by project or by class would help me as well.

Thanks!

**Comment by rsmckinney** [comment]
Fix available in release `2019.1.23`

Thanks for reporting this!

**Comment by icager** [comment]
Thanks for getting to this so quickly! Just wanted to confirm that I've tried 2019.1.23 in my use case and it's working fine. Thanks again!
